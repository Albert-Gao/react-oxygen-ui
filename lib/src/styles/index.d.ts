import { Responsive } from '../components';
import { defaultTheme } from './theme/theme';
export declare type ThemeType = typeof defaultTheme;
export declare type SpaceType = Responsive;
export declare type FontSizeType = Responsive;
export declare type GradientType = keyof ThemeType['gradients'];
export declare type ColorType = string | string[];
export declare type FontWeightType = keyof ThemeType['fontWeights'];
export declare type BoxShadowType = number | number[];
export { DEFAULT_FONT_SIZE, fontSpecs } from './theme/constants/fontSpecs';
export { DEFAULT_SPACE } from './theme/constants/spaces';
export { DEFAULT_SIZE } from './theme/constants/sizes';
export { DEFAULT_COLORS } from './theme/constants/colors';
export { defaultTheme } from './theme/theme';
export * from './theme/theme.utils';
export { styled, css, createGlobalStyle, keyframes, ThemeProvider, } from './styled';
export { getTransitionValue, getSafeValue, getSafeCss, } from './utils';
