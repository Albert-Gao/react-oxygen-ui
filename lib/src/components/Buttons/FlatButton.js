"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const styles_1 = require("../../styles");
const utils_1 = require("../../utils/utils");
const BaseBox_1 = require("../Box/BaseBox");
const shared_1 = require("./shared");
exports.FlatButton = ({ noDefaultHoverCSS, children, css, takeRef, ...rest }) => (<BaseBox_1.BaseBox as="a" fontSize="inherit" px={styles_1.DEFAULT_SPACE.xxSmall} color="primary" css={`
      ${shared_1.getBaseButtonCSS(noDefaultHoverCSS)};
      ${styles_1.getSafeCss(css)};
    `} {...utils_1.getRefProp(takeRef)} {...rest}>
    {children}
  </BaseBox_1.BaseBox>);
