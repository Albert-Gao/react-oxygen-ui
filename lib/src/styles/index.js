"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
var fontSpecs_1 = require("./theme/constants/fontSpecs");
exports.DEFAULT_FONT_SIZE = fontSpecs_1.DEFAULT_FONT_SIZE;
exports.fontSpecs = fontSpecs_1.fontSpecs;
var spaces_1 = require("./theme/constants/spaces");
exports.DEFAULT_SPACE = spaces_1.DEFAULT_SPACE;
var sizes_1 = require("./theme/constants/sizes");
exports.DEFAULT_SIZE = sizes_1.DEFAULT_SIZE;
var colors_1 = require("./theme/constants/colors");
exports.DEFAULT_COLORS = colors_1.DEFAULT_COLORS;
var theme_1 = require("./theme/theme");
exports.defaultTheme = theme_1.defaultTheme;
__export(require("./theme/theme.utils"));
var styled_1 = require("./styled");
exports.styled = styled_1.styled;
exports.css = styled_1.css;
exports.createGlobalStyle = styled_1.createGlobalStyle;
exports.keyframes = styled_1.keyframes;
exports.ThemeProvider = styled_1.ThemeProvider;
var utils_1 = require("./utils");
exports.getTransitionValue = utils_1.getTransitionValue;
exports.getSafeValue = utils_1.getSafeValue;
exports.getSafeCss = utils_1.getSafeCss;
