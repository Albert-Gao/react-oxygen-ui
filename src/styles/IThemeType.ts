//
// The following are the expected properties for your styled-components' theme object
// styled-system depends on them to provide you all the conveniences
//
// [Key]	        [Type]	          [Description]
// breakpoints	  Array	            Array of viewport widths to use for min-width media queries
// fontSizes	    Array	            Array of numbers to use as a typographic scale
// colors	        Object	          Color names that can be used in color, bg, and borderColor props
// space	        Array	            Array of numbers for use as margin and pixel values
// fonts	        Array or Object	  Values for the fontFamily prop
// fontWeights	  Array or Object	  Values for fontWeight prop
// lineHeights	  Array or Object	  Values for lineHeight prop
// letterSpacings	Array or Object	  Values for letterSpacing prop
// shadows	      Array or Object	  Values for boxShadow prop
// borders	      Array or Object	  Values for border props
// radii	        Array or Object	  Values for borderRadius props
// opacity	      Array or Object	  Values for opacity props

type NumberOrString = number | string;

type ObjectOrArray<T> =
  | {
      [propName: string]: T;
    }
  | T[];

export interface ITheme {
  breakpoints: NumberOrString[];
  fontSizes: NumberOrString[];
  colors: { [colorName: string]: string };
  space: NumberOrString[];
  fontWeights: ObjectOrArray<number>;
  lineHeights?: ObjectOrArray<NumberOrString>;
  borders: ObjectOrArray<NumberOrString>;
  radii: ObjectOrArray<string>;
  shadows: ObjectOrArray<string>;
  opacity: ObjectOrArray<string>;
  gradients?: object;
}

export const defaultTheme: ITheme = {
  breakpoints: ['40em', '52em', '64em'],
  fontSizes: ['12px', '14px', '16px', '24px', '30px', '36px', '48px'],
  colors: {
    white: 'white',
  },
  fontWeights: [100, 300, 400, 500, 700, 900],
  borders: [0, '1px solid'],
  radii: ['4px'],
  shadows: [
    '0px 1px 2px rgba(0,0,0,0.20)',
    '0px 2px 4px rgba(0,0,0,0.20)',
    '0px 4px 8px rgba(0,0,0,0.20)',
    '0px 8px 16px rgba(0,0,0,0.20)',
    '0px 12px 24px rgba(0,0,0,0.20)',
  ],
  opacity: [],
  space: [0, 4, 8, 16, 24, 32, 48, 64, 96, 128, 256, 512],
};
