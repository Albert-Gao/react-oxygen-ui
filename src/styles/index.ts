import { Responsive } from '../components';
import { defaultTheme } from './theme/theme';


export type ThemeType = typeof defaultTheme;
export type SpaceType = Responsive;
export type FontSizeType = Responsive;
export type GradientType = keyof ThemeType['gradients'];
export type ColorType = string | string[];
export type FontWeightType = keyof ThemeType['fontWeights'];
export type BoxShadowType = number | number[];

export { DEFAULT_FONT_SIZE, fontSpecs } from './theme/constants/fontSpecs';
export { DEFAULT_SPACE } from './theme/constants/spaces';
export { DEFAULT_SIZE } from './theme/constants/sizes';
export { DEFAULT_COLORS } from './theme/constants/colors';
export { defaultTheme } from './theme/theme';
export * from './theme/theme.utils';

export {
  styled,
  css,
  createGlobalStyle,
  keyframes,
  ThemeProvider,
} from './styled';

export {
  getTransitionValue,
  getSafeValue,
  getSafeCss,
} from './utils';
