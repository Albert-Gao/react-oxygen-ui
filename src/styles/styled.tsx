import React from 'react';
import * as styledComponents from 'styled-components';
// tslint:disable-next-line:no-duplicate-imports
import { ThemedStyledComponentsModule } from 'styled-components';
import { ThemeType } from './index';
import { defaultTheme, ITheme } from './theme/theme';

const ThemeProvider: React.FC<{
  theme: ITheme;
  // tslint:disable-next-line:no-any
  children: React.ReactElement<any>;
}> = ({ theme, children }) => (
  <styledComponents.ThemeProvider theme={theme || defaultTheme}>
    {children}
  </styledComponents.ThemeProvider>
);

const {
  default: styled,
  css,
  createGlobalStyle,
  keyframes,
} = styledComponents as ThemedStyledComponentsModule<ThemeType>;

export { styled, css, createGlobalStyle, keyframes, ThemeProvider };
