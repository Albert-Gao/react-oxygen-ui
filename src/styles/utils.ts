import { FlattenInterpolation } from 'styled-components';
import { createMediaQuery } from 'styled-system';
import { CSSType } from '../components/Box/BaseBox.type';
import { is } from '../utils/utils';
import { ITheme } from './IThemeType';

export const getMediaQuery = (
  screenSize: number | string,
  css: string | FlattenInterpolation<ITheme>,
) =>
  `${createMediaQuery(screenSize)} {
    ${css};
  }`;

export const getTransitionValue = (propertyNames: string = 'transform') =>
  `${propertyNames} 0.3s cubic-bezier(0.53, 0.01, 0.36, 1.63);`;

export function getSafeValue<T>(target: unknown, defaultValue: T) {
  return is.notExist(target) ? defaultValue : target;
}

export function getSafeCss(target?: CSSType) {
  return getSafeValue(target, '') as CSSType;
}
