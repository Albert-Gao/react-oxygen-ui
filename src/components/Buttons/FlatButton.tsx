import React from 'react';
import { DEFAULT_SPACE, getSafeCss } from '../../styles';
import { getRefProp } from '../../utils/utils';
import { BaseBox } from '../Box/BaseBox';
import { IBaseBoxWithTakeRef } from '../Box/BaseBox.type';
import { IButton } from './Button';
import { getBaseButtonCSS, INoDefaultHoverCSS } from './shared';

export interface IFlatButton
  extends IBaseBoxWithTakeRef,
    INoDefaultHoverCSS,
    IButton {}

export const FlatButton: React.FC<IFlatButton> = ({
  noDefaultHoverCSS,
  children,
  css,
  takeRef,
  ...rest
}) => (
  <BaseBox
    as="a"
    fontSize="inherit"
    px={DEFAULT_SPACE.xxSmall}
    color="primary"
    css={`
      ${getBaseButtonCSS(noDefaultHoverCSS)};
      ${getSafeCss(css)};
    `}
    {...getRefProp(takeRef)}
    {...rest}
  >
    {children}
  </BaseBox>
);
