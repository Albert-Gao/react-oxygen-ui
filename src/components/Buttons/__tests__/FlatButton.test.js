import 'jest-styled-components';
import 'jest-dom/extend-expect';
import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { FlatButton } from '../FlatButton';
import { ThemeProvider } from '../../../styles';

afterEach(cleanup);

describe('Test <FlatButton />: ', () => {
  test('css prop should be added', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <FlatButton css={'background-color: black;'} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();

    // the new css is there
    expect(firstChild).toHaveStyleRule('background-color', 'black');

    // the old css has been merged and not lost
    expect(firstChild).toHaveStyleRule('white-space', 'nowrap');
  });

  test('[FlatButton] undefined css prop should be ignored', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <FlatButton css={undefined} />
      </ThemeProvider>,
    ).container;

    // invalid css prop will cause `undefined: ;` in result
    // which is invalid
    // so it won't pass this assertion
    expect(firstChild).toMatchSnapshot();
  });
});
