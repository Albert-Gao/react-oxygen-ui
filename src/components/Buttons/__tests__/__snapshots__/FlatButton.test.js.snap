// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FlatButton />:  [FlatButton] undefined css prop should be ignored 1`] = `
.c0 {
  box-sizing: border-box;
  padding-left: 16px;
  padding-right: 16px;
  font-size: inherit;
  color: #2CAACA;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  display: inline-block;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-align: center;
  line-height: inherit;
  white-space: nowrap;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.9;
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

.c0:active {
  -webkit-transform: scale(1) translate(1px,1px);
  -ms-transform: scale(1) translate(1px,1px);
  transform: scale(1) translate(1px,1px);
}

<a
  class="c0"
  color="primary"
  font-size="inherit"
/>
`;

exports[`Test <FlatButton />:  css prop should be added 1`] = `
.c0 {
  box-sizing: border-box;
  padding-left: 16px;
  padding-right: 16px;
  font-size: inherit;
  color: #2CAACA;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  display: inline-block;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-align: center;
  line-height: inherit;
  white-space: nowrap;
  cursor: pointer;
  background-color: black;
}

.c0:hover {
  opacity: 0.9;
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

.c0:active {
  -webkit-transform: scale(1) translate(1px,1px);
  -ms-transform: scale(1) translate(1px,1px);
  transform: scale(1) translate(1px,1px);
}

<a
  class="c0"
  color="primary"
  font-size="inherit"
/>
`;
